ctx.go

```diff
@@ -0,0 +1,28 @@
+package clickhouse
+
+import (
+	"context"
+	"net/http"
+)
+
+type ctxKey uint8
+
+const (
+	ctxRespCallbackKey ctxKey = iota + 1
+)
+
+// RespCallback is a transport response callback.
+type RespCallback func(*http.Request, *http.Response) error
```

Comment:
RespCallback is short for a response callback, nothing stated about the request. Maybe we should call it TransportCallback